redpanda/templates/configmap.yaml

```diff
@@ -30,6 +30,21 @@ metadata:
   {{- toYaml . | nindent 4 }}
 {{- end }}
 data:
+{{- if (include "redpanda.semver" . | semverCompare ">=22.1.4") }}
+  bootstrap.yaml: |
+{{- if .Values.auth.sasl.enabled }}
+      enable_sasl: true 
+  {{- with .Values.auth.sasl.users }}
+      superusers: ['{{ (first .).name }}']
+  {{- end }}
+{{- end }}
+ {{- with .Values.config.cluster }}
+    {{- toYaml . | nindent 6 }}
+  {{- end }}
+  {{- with .Values.config.tunable }}
+    {{- toYaml . | nindent 6 }}
+  {{- end }}
+{{- end }}
```

Comment:
I think there should be an `else` here, since we would want to take a different path for lower semver values (Redpanda versions without central config support). In this new `else` branch, `.Values.config.cluster` and `.Values.config.tunable` would be written to `redpanda.yaml` in the appropriate place.

Reply:
Its pulling the old config format through lines 64-66

Reply:
What this means is that if you have an older version of the values.yaml format, it is implied that you have transformed it to the cluster/tunables config format before you upgrade. If its lower than 22.1.1 then the old format should work.

Reply:
We need to think about documenting the upgrade paths .......

Reply:
We discussed this internally giving two options: 1) one `values.yaml` that works across all Redpanda versions and 2) requiring the user to specify an older Redpanda version as well as properly format the `config` section for that version.

We decided to go with option 2 since this will be less code in the helm chart (and potential for issues down the road), and it would be a more deliberate path for the user to choose an older version (considering we would rather support new Redpanda versions). We will provide documentation on using older Redpanda versions with the latest helm chart, which should help users with this.